### PR TITLE
Moving firehose part to a consumable module

### PR DESCRIPTION
**Pull request by rporres**

### CODE REVIEW

File: modules/splunk-firehose-connection/variables.tf [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  type        = string
+}
+
+variable "tags" {
+  description = "A map of tags to add to all resources"
+  type        = map(string)
+  default     = {}
+}
+
+### TODO: order this
```
Comment (eisraeli) [inline modules/splunk-firehose-connection/variables.tf]: What have you meant to do here ?

Comment (rporres) [reply modules/splunk-firehose-connection/variables.tf]: variables are ordered alphabetically :)

File: modules/splunk-firehose-connection/variables.tf [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  description = "The amount of time, in seconds between 180 and 600, that Kinesis Firehose waits to receive an acknowledgment from Splunk after it sends it data."
+  type        = string
+  default     = 300
+}
+
+variable "hec_token" {
+  description = "Pass the HEC token in plain text (not recommended) or through a parameter store, through a KMS encryption module, etc..."
+  type        = string
+  sensitive   = true
+}
```
Comment (eisraeli) [inline modules/splunk-firehose-connection/variables.tf:23]: We should make it as `required = true`

File: modules/splunk-firehose-connection/variables.tf [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+variable "s3_prefix" {
+  description = "Prefix for the kinesis firehose related bucket."
+  type        = string
+  default     = "kinesis-firehose/"
+}
+
+variable "splunk_endpoint" {
+  description = "Splunk endpoint"
+  type        = string
+}
```
Comment (eisraeli) [inline modules/splunk-firehose-connection/variables.tf:69]: We should define it as `required = true`

File: modules/splunk-firehose-connection/variables.tf [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+variable "tags" {
+  description = "A map of tags to add to all resources"
+  type        = map(string)
+  default     = {}
+}
+
+### TODO: order this
+variable "name" {
+  type = string
+}
```
Comment (eisraeli) [inline modules/splunk-firehose-connection/variables.tf:80]: We should define it as `required = true`

File: modules/splunk-firehose-connection/variables.tf [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+variable "cloudwatch_log_group_prefix" {
+  type = string
+}
```
Comment (eisraeli) [inline modules/splunk-firehose-connection/variables.tf:84]: This should either have a default or be marked as `requried = true` as well.

**Pull request merged**